internal/statemachine/common_states.go

```diff
@@ -138,29 +139,72 @@ func (stateMachine *StateMachine) generateDiskInfo() error {
 // on a 100MiB filesystem, ext4 takes a little over 7MiB for the
 // metadata. Use 8MB as a minimum padding here
 func (stateMachine *StateMachine) calculateRootfsSize() error {
-	rootfsSize, err := helper.Du(stateMachine.tempDirs.rootfs)
-	if err != nil {
-		return fmt.Errorf("Error getting rootfs size: %s", err.Error())
-	}
-	var rootfsQuantity quantity.Size = rootfsSize
-	rootfsPadding := 8 * quantity.SizeMiB
+	if stateMachine.commonFlags.Size != "" {
+		var parsedSize quantity.Size
+
+		// identify which structure has the rootfs
+		var rootfsVolume *gadget.Volume
+		var rootfsVolumeName string
+		for volumeName, volume := range stateMachine.GadgetInfo.Volumes {
+			for _, structure := range volume.Structure {
+				if structure.Size == 0 {
```

Comment:
Okay, guess the writable partition is the only partition we allow to not have a size given, right? I guess yes, so this is fine. Maybe since we only expect one partition like this, shouldn't we `break` when we find it?

Reply:
Yes, any other partition not having a size specified will have already failed while loading the gadget YAML. Good point on the break statement, I'll add one.